spyder_kernels/console/kernel.py

```diff
@@ -124,6 +125,26 @@ def frontend_call(self, blocking=False, broadcast=True,
             callback=callback,
             timeout=timeout)
 
+    def get_state(self):
+        """"get current state to send to the frontend"""
+        state = {}
+        with WriteContext("get_state"):
+            if self._cwd_initialised:
+                state["cwd"] = self.get_cwd()
+            state["namespace_view"] = self.get_namespace_view()
+            state["var_properties"] = self.get_var_properties()
```

Comment:
It is safer to know what the var_properties are that have been used, in case the kernel is out of sync for some reason. But I don't have a strong oppinion on that.

Reply:
Are you sure? `def get_var_properties(self):` loops on the variables in the namespace, so it looks like this is not information the frontend has.

Reply:
You're totally, sorry for the noise. I confused this functionality with what the `set_namespace_view_settings` method does.